# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/webkitgtk/x32.patch
# Copyright (C) 2022 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- webkitgtk-2.36.0/Source/WTF/wtf/PlatformCPU.h.vanilla	2022-04-23 23:20:12.437098667 +0200
+++ webkitgtk-2.36.0/Source/WTF/wtf/PlatformCPU.h	2022-04-23 23:22:43.065092661 +0200
@@ -112,8 +112,12 @@
 /* CPU(X86_64) - AMD64 / Intel64 / x86_64 64-bit */
 #if   defined(__x86_64__) \
     || defined(_M_X64)
+#ifdef __ILP32__
+#define WTF_CPU_X86_64_32 1
+#else
 #define WTF_CPU_X86_64 1
 #define WTF_CPU_X86_SSE2 1
 #define WTF_CPU_KNOWN 1
+#endif
 #endif
 
--- webkitgtk-2.36.0/Source/WTF/wtf/PageBlock.h.vanilla	2022-04-23 23:24:32.510088297 +0200
+++ webkitgtk-2.36.0/Source/WTF/wtf/PageBlock.h	2022-04-23 23:29:51.547075576 +0200
@@ -49,7 +49,7 @@
 constexpr size_t CeilingOnPageSize = 16 * KB;
 #elif CPU(PPC) || CPU(PPC64) || CPU(PPC64LE) || CPU(UNKNOWN)
 constexpr size_t CeilingOnPageSize = 64 * KB;
-#elif OS(WINDOWS) || CPU(X86) || CPU(X86_64) || CPU(ARM) || CPU(ARM64) || CPU(RISCV64)
+#elif OS(WINDOWS) || CPU(X86) || CPU(X86_64) || CPU(X86_64_32) || CPU(ARM) || CPU(ARM64) || CPU(RISCV64)
 constexpr size_t CeilingOnPageSize = 4 * KB;
 #else
 #error Must set CeilingOnPageSize in PageBlock.h when adding a new CPU architecture!
